File: kuba-front/src/app/shared/models/incident-response.ts
import IncidentType from "../enums/incident-type.enum";
import SeverityType from "../enums/severity.enum";

export default interface IncidentResponse {
    id: string;
    title: string;
    description: string;
    reportedDate: Date;
    severityType: SeverityType;
    incidentType: IncidentType;
    createdAt: Date;
    updatedAt: Date | null;
}

File: kuba-back/Kuba.Api/Kuba.Api/Dtos/Incident/IncidentResponse.cs
using Kuba.Domain.Enums;

namespace Kuba.Api.Dtos.Incident
{
    public class IncidentResponse
    {
        public Guid Id { get; set; }
        public string Title { get; set; } = string.Empty;
        public string Description { get; set; } = string.Empty;
        public DateTime ReportedDate { get; set; }
        public SeverityType SeverityType { get; set; }
        public IncidentType IncidentType { get; set; }

        public DateTime CreatedAt { get; set; }
        public DateTime? UpdatedAt { get; set; }
    }
}
